src/serialization.ts

```diff
@@ -428,3 +430,60 @@ export function serializeCredentialDeploymentTransactionForSubmission(
     );
     return Buffer.from(credentialDeploymentInfo.serializedTransaction, 'hex');
 }
+
+/**
+ *
+ * @param contractName name of the contract to init contract parameters
+ * @param userInput  user input
+ * @param modulefileBuffer buffer of embedded schema file
+ * @returns
+ */
+export function serializeInitContractParameters(
+    contractName: string,
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any, @typescript-eslint/explicit-module-boundary-types
+    userInput: any,
+    modulefileBuffer?: globalThis.Buffer
+): Buffer {
+    const getSchemaModule = deserialModuleFromBuffer(modulefileBuffer);
+    if (getSchemaModule !== undefined) {
```

Comment:
Why would the `schemaModule` be undefined here?
And if it is, should we not throw an error instead of returning an empty buffer?

Reply:
@shjortConcordium It can be undefined if we provide a schema file which is not a valid. In this case we will throw an error Schema module not found. Please provide a valid schema file.